strategy/strategy.go

```diff
@@ -0,0 +1,52 @@
+package strategy
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Strategy describes a collection of metrics built with a struct, allowing any multi-level
+// combination of the 4 Prometheus metric types and other Strategies.
+// e.g. RED - describes a set of metrics that work well for monitoring
+// request-handling services.
+//
+//	type RED struct {
+//		Requests *prometheus.CounterVec
+//		Errors *prometheus.CounterVec
+//		Duration *prometheus.HistogramVec
+//	}
+type Strategy interface {
+	Register() error
```

Comment:
Would we want a `Validate` method being added as part of the interface? My reasoning is, how are we enforcing that users of the library are actually inputting the `Opts` struct properly? What if people are inputting empty strings in some crucial fields like Label names and etc?

Reply:
I believe that's what the default in the switch in ``RegisterStrategyFields`` does. But perhaps a validate method would be a good idea.

Reply:
@nothingto and @ajk314 I'll sketch up a `Validate` POC in a subsequent PR.

Reply:
[@nothingto](https://github.com/nothingto) and [@ajk314](https://github.com/ajk314) I'll sketch up a Validate POC in a subsequent PR.